src/index.ts

```diff
@@ -1,93 +1,215 @@
+type Lecturer = {
+  name: string
+  surname: string
+  position: string
+  company: string
+  experiance: string[]
+  courses: string[]
+  contacts: (string | number)[]
+}
+
 class School {
   // implement 'add area', 'remove area', 'add lecturer', and 'remove lecturer' methods
+  _areas: Area[] = [];
+  _lecturers: Lecturer[] = []; // Name, surname, position, company, experience, courses, contacts
+
+  get areas(): Area[] {
+    return this._areas;
+  }
 
-  _areas = [];
-  _lecturers = []; // Name, surname, position, company, experience, courses, contacts
+  get lecturers(): Lecturer[] {
+    return this._lecturers;
+  }
+
+  addArea(areaToAdd: Area): void {
+    this._areas.push(areaToAdd)
+  }
 
-  get areas() {
-    this._areas;
+  removeArea(areaToRemove: Area): void {
+    this._areas = this._areas.filter((area: Area): boolean => area !== areaToRemove)
```

Comment:
It is not correct to compare two objects, such as "Area", directly. Instead, it is better to compare objects by name.